app/src/main/java/com/github/se/cyrcle/model/user/UserViewModel.kt

```diff
@@ -153,6 +184,68 @@ class UserViewModel(
     }
   }
 
+  /**
+   * Upload the picture to the cloud storage update the user object in the database with the cloud
+   * path created. update the user object in the viewmodel with the url of the image and the cloud
+   * path
+   */
+  private fun uploadProfilePicture(context: Context) {
+    if (imageRepository == null) {
+      Log.e("UserViewModel", "ImageRepository is null")
+      return
+    }
+    val user = currentUser.value ?: return
+    val fileUri = user.localSession?.profilePictureUri ?: return
+    val destinationPath = "profile_pictures/${user.public.userId}"
+    imageRepository.uploadImage(
+        context = context,
+        fileUri = fileUri,
+        destinationPath = destinationPath,
+        onSuccess = { url ->
+          Log.d("UserViewModel", "Image uploaded successfully to $url")
+          val updatedUser =
+              user.copy(
+                  public = user.public.copy(profilePictureCloudPath = destinationPath),
+                  localSession =
+                      user.localSession.copy(profilePictureUri = null, profilePictureUrl = url))
+          updateUser(updatedUser)
+        },
+        onFailure = { Log.e("UserViewModel", "Failed to upload image") })
+  }
+
+  /**
+   * Updates the user object with the profile picture url fetched from the repo.
+   *
+   * @param user the user object to update (doesn't have to be the current user)
+   * @param onSuccess the callback to call with the updated user
+   */
+  private fun transformPathToUrl(user: User, onSuccess: (User) -> Unit, onFailure: () -> Unit) {
```

Comment:
you forgot to document the `onFailure` here

Reply:
Solved